### DISCUSSION TITLE
Is it possible to use the colors as background

### DISCUSSION THREAD

**Discussion by FeliNUx42** [danwild/leaflet-velocity#91] [Q&A]
Hello everyone,
I recently discovered leaflet-velocity and I'd like to put the colors of the strength of the wind as a background.
Is this possible or how would you do it?

I'd like the result to look like that:
![image](https://user-images.githubusercontent.com/76401024/183300716-c44ca799-d8bc-4fac-b64e-fdd1aab58a8c.png)

Thanks...

**Comment by danwild** [comment]
Hey @FeliNUx42,

A similar question has previously been raised in https://github.com/onaci/leaflet-velocity/issues/19.
As I noted in that issue - I don't have time to implement this feature, but I'm open to PR's.

Alternatively, I think this could also be achieved by simply adding magnitudes data as a different layer.
If you look at the code for windy.com that is actually how they do it (the animated particles are a different layer to the 'background' layer of scalar magnitudes).

Cheers,
Dan

**Reply by cackland** [reply]
I'm trying to implement this sort of coloured background layer. I have tried a few leaflet heat map type options - but don't really show the colours consistently during map interaction (see https://live-wind.com.au). Am interested to know @danwild what you mean by "magnitudes data as a different layer."?

**Reply by danwild** [reply]
Heatmaps are typically used to represent density, not magnitude over a geographic area.

Depending on your use-case, there are 2x main options @cackland:

## 1. Use a server 

Use a web service such as GeoServer to publish raster (e.g. wms) or vector data (vector tiles), then consume them in your client app.

## 2. Client-side rendering

Just render the data on the client directly, e.g. I wrote some other plugins to help with this: 

- https://github.com/onaci/Leaflet.glify.layer
- https://github.com/onaci/leaflet-geotiff-2

The option you choose depends largely on the format and size of your data (large or complex data => server-side), or the typical user (e.g. limited bandwidth => server-side).

**Reply by cackland** [reply]
Thanks @danwild  - will have a look into this. Appreciate the info. Kind of lines up with what I was thinking.
I was doing a bit of reading up - I'm using griddata to interpolate a bunch of irregular points into a grid for my wind animation as well as the 'magninute' values of wind. I also thought of writing out a raster (https://geophysics.in/how-to-convert-point-data-to-raster-file/) using something like rasterio, and then slicing into leaflet/map tiles using something like gdal2tiles. Will have a look at your plugins.
